tests/test_metrics/test_intervals_metrics.py

```diff
@@ -190,3 +289,103 @@ def test_using_not_presented_quantiles(metric, tsdataset_with_zero_width_quantil
 )
 def test_metrics_greater_is_better(metric, greater_is_better):
     assert metric.greater_is_better == greater_is_better
+
+
+@pytest.mark.parametrize(
+    "metric_class",
+    (Coverage, Width),
+)
+def test_missing_values_in_pred_error(metric_class, tsdataset_with_intervals_and_missing_values):
+    true_ts, forecast_ts = tsdataset_with_intervals_and_missing_values
+    metric = metric_class()
+    with pytest.raises(ValueError, match="There are NaNs in y_pred"):
+        _ = metric(y_true=true_ts, y_pred=forecast_ts)
+
+
+@pytest.mark.parametrize(
+    "metric",
+    (
+        Coverage(missing_mode="error"),
+        Width(missing_mode="error"),
+    ),
+)
+def test_missing_values_in_true_error(metric, tsdataset_with_intervals_and_missing_values):
+    forecast_ts, true_ts = tsdataset_with_intervals_and_missing_values
+    with pytest.raises(ValueError, match="There are NaNs in y_true"):
+        _ = metric(y_true=true_ts, y_pred=forecast_ts)
+
+
+@pytest.mark.parametrize(
+    "metric, expected_type",
+    (
+        (Coverage(mode="per-segment", missing_mode="ignore"), type(None)),
+        (Width(mode="per-segment", missing_mode="ignore"), float),
+    ),
+)
+@pytest.mark.parametrize(
+    "dataset_name, empty_segment",
+    (
+        ("tsdataset_with_intervals_and_missing_values", None),
+        ("tsdataset_with_intervals_and_missing_segment", "segment_1"),
+    ),
+)
+def test_missing_values_ignore_per_segment(metric, dataset_name, empty_segment, expected_type, request):
+    forecast_ts, true_ts = request.getfixturevalue(dataset_name)
+    segments = set(forecast_ts.segments)
+
+    value = metric(y_true=true_ts, y_pred=forecast_ts)
+
+    assert isinstance(value, dict)
+    assert value.keys() == segments
+
+    if empty_segment is not None:
+        assert isinstance(value[empty_segment], expected_type)
+        value.pop(empty_segment)
+
+    assert all(isinstance(cur_value, float) for cur_value in value.values())
+
+
+@pytest.mark.parametrize(
+    "metric",
+    (
+        Coverage(mode="macro", missing_mode="ignore"),
+        Width(mode="macro", missing_mode="ignore"),
+    ),
+)
+@pytest.mark.parametrize(
+    "dataset_name", ("tsdataset_with_intervals_and_missing_values", "tsdataset_with_intervals_and_missing_segment")
+)
+def test_segment_all_missing_ignore_macro(metric, dataset_name, request):
```

Comment:
`test_segment_all_missing_ignore_macro` -> `test_segment_all_missing_values_ignore_macro`?

Reply:
This is a small, so it could be skipped.